### ISSUE TITLE
Log Node.js version using

### ISSUE THREAD

**Issue by fishcharlie** [actions/setup-node#90]
Couple of things.

First if you specify `node-version` of `10`, does that use the absoute latest version of Node.js v10 (both minor and patch versions)?

Second, does `node-version` of `10.x` use the latest version when taking account patch versions? Or does `.x` just mean the minor version?

Finally, it'd be great if this action printed the Node.js version it decided to use in the logs of GitHub Actions so developers can determine which Node.js version it decided to use, which would make debugging SO much easier.

<img width="1055" alt="Screen Shot 2019-11-22 at 3 49 46 PM" src="https://user-images.githubusercontent.com/860375/69465889-bb052a00-0d3f-11ea-8f40-41f5e163aecf.png">

**Comment by joshmgross** [comment]
👋 Hey @fishcharlie. Specifying a `node-version` of `10` and `10.x` will have the same result. This follows [semver](https://semver.org/), where `10 == 10.x == 10.x.x`, which is effectively "the latest version of Node.js v10 taking into account both minor and patch versions. 

>Finally, it'd be great if this action printed the Node.js version it decided to use in the logs of GitHub Actions

Great idea! That should be a quick fix.

**Comment by bryanmacfarlane** [comment]
Yes, it should work with any semver version spec.  As part of this issue, I'll make sure it's logged as well.  I'll also update the readme / docs

**Comment by fishcharlie** [comment]
Any updates on this? cc/ @bryanmacfarlane @joshmgross

**Comment by joshmgross** [comment]
@bryanmacfarlane are you still planning on doing this? If not,  I can pick this up.

**Comment by bryanmacfarlane** [comment]
@joshmgross - I'm fine with you picking this up.  If you do, look at recent commits in setup-go where I did it.  I think it should also output npm version.

**Comment by fishcharlie** [comment]
Outputting the npm version too would be amazing!!!

**Comment by demian85** [comment]
Can someone please explain why my node version is not 12 as specified?

Part of my config file is
```
jobs:
  primary:
    runs-on: ubuntu-latest
    strategy:
      matrix:
        node-version: [12.x]

    steps:
      - uses: actions/checkout@v1

      - name: Cache node modules
        uses: actions/cache@v1.1.0
        with:
          path: ~/.npm
          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
          restore-keys: |
            ${{ runner.os }}-node-
      - name: Node ${{ matrix.node-version }}
        uses: actions/setup-node@v1.1.0
        with:
          node-version: ${{ matrix.node-version }}

      - name: Installing dependencies
        run: npm install
``` 

And logging the node version to the console results in `node version: 10.20.1` 

I do not quite understand where is the problem.

**Comment by bryanmacfarlane** [comment]
@demian85 can you log a separate issue?  I would also try setup-node@v1 to ensure you're getting latest along with bug fixes 

Here's a workflow of mind doing @ 12.x and it leads to node 12. https://github.com/bryanmacfarlane/actions-playground/runs/570652210?check_suite_focus=true

**Comment by fishcharlie** [comment]
@joshmgross Would be more than happy to review and/or test a PR for this when you have it up! Thanks so much for helping with this.

**Comment by joshmgross** [comment]
PR for this here: https://github.com/actions/setup-node/pull/137

**Comment by EdJoPaTo** [comment]
This was resolved via #137 but its not working anymore?

Log from the used action isnt showing the specific NPM version:
```plaintext
Run actions/setup-node@v2
  with:
    node-version: 16
    always-auth: false
    check-latest: false
    token: ***
Attempting to download 16...
Acquiring 16.11.1 - x64 from https://github.com/actions/node-versions/releases/download/16.11.1-1336322549/node-16.11.1-linux-x64.tar.gz
Extracting ...
/usr/bin/tar xz --strip 1 --warning=no-unknown-keyword -C /home/runner/work/_temp/192ce410-4dad-49a1-99ac-d942052f626f -f /home/runner/work/_temp/5a93bf66-8337-45eb-b047-0b5bb8eab63e
Adding to the cache ...
Done
```

**Comment by joshmgross** [comment]
Looks like the changes added in #137 were inadvertently removed in #147

**Comment by dmitry-shibanov** [comment]
Hello everyone. We've released [a new version 3.5.1](https://github.com/actions/setup-node/releases/tag/v3.5.1) with logging Nodejs version. For now we didn't update the major tag. Could you please confirm that everything works as expected ?

**Comment by dmitry-shibanov** [comment]
Hello everyone. For now I'm going to close the issue. If you have any concerns feel free to pings us.
